### PR TITLE
Admin item create 98; closes #98

### LINKED ISSUE #98: As an admin, I can create item listings

### PR DESCRIPTION
**Pull request by nathanows**
Added the ability to create items from /admin/items (a link on that page points to the new item route).

The one facet not tested is that loaded images show up on the page. We can revisit this with some help hopefully during the week, but basically when the images are uploaded their names are randomized so i'm not sure what to test against (in the guest view integration test i was pointing to existing images so i was able verify that the img title was equal to the one created).

NOTE: If you're adding images, it will create them in the public/system/images file for each image loaded. This seems like natural functionality as you're actually adding them to the database. Unfortunately yolo'ing the DB doesnt remove these... not sure exactly how this should be handled (and it seems like it would be fine in production), but I just went in and deleted all of the folders in public/system/images/imgs from 005 to the end (there are 4 images currently used in our seed data that should be kept).

### CODE REVIEW

File: app/views/admin/items/_form.html.erb [hunk 1]
```diff
@@ -0,0 +1,20 @@
+<%= bootstrap_form_for @item, :html => { multipart: true }  do |f| %>
+  <%= f.text_field :title %>
+  <%= f.text_area :description %>
+  <%= f.number_field :unit_price, label: "Price (in cents)"%>
```
Comment (EmilyMB) [inline app/views/admin/items/_form.html.erb:4]: Maybe this should be label: "Price"? (and calculate the :unit_price on the backend)?

File: app/views/admin/shared/_admin_menu.html.erb [hunk 2]
```diff
@@ -3,7 +3,7 @@
     <h3 class="admin-panel-title">Admin Panel</h3>
     <hr />
     <li><%= link_to "Dashboard", admin_path %></li>
-    <li><%= link_to "Manage Items", "#" %></li>
+    <li><%= link_to "Manage Items", items_path %></li>
```
Comment (EmilyMB) [inline app/views/admin/shared/_admin_menu.html.erb:6]: :tada: always better to have a real path than a "#" :smile:

File: spec/integration/admin_items_spec.rb [hunk 3]
```diff
@@ -6,7 +6,6 @@
 
     it "an admin can see an item listing page" do
       mock_admin
-
```
Comment (EmilyMB) [inline spec/integration/admin_items_spec.rb:9]: Maybe you want to leave the blank line in there? To keep the separation between setup (mock_admin) and exercise (visit items_path).

File: spec/integration/admin_items_spec.rb [hunk 4]
```diff
@@ -16,7 +15,6 @@
       mock_admin
       image = create(:image)
       item = create(:item, title: "A Pizza Pie", image_id: image.id)
-
```
Comment (EmilyMB) [inline spec/integration/admin_items_spec.rb:19]: Same thought as above.

File: spec/integration/admin_items_spec.rb [hunk 5]
```diff
@@ -25,10 +23,58 @@
     end
   end
 
+  describe "creating an item" do
+
+    it "an admin sees a create button" do
```
Comment (EmilyMB) [inline spec/integration/admin_items_spec.rb]: Maybe it "shows a create button"?

File: spec/integration/admin_items_spec.rb [hunk 6]
```diff
@@ -25,10 +23,58 @@
...
+  private
+
   def mock_admin
     admin = create(:admin)
     allow_any_instance_of(ApplicationController).
       to receive(:current_user).
       and_return(admin)
   end
+
+  def fill_in_new_item
```
Comment (EmilyMB) [inline spec/integration/admin_items_spec.rb:74]: :+1: Nice way to pull out the logic into separate methods

File: app/views/admin/items/_form.html.erb (hunk 1 above)
Comment (nathanows) [inline app/views/admin/items/_form.html.erb:4]: Yeah, the display name on the field is 'Price (in cents)', and it maps to unit_price in the background. I dont love that you have to enter it in cents right now, but was going to be a good amount of work to make it work in dollar format

File: spec/integration/admin_items_spec.rb (hunk 3 above)
Comment (nathanows) [inline spec/integration/admin_items_spec.rb:9]: yeah... i had some internal conflict on that. I landed on the side of `visit items_path` being a setup action. its how you set the test up to start is how i thought of it. Its not clicking on anything or filling anything in, just prepping the start of the test. I dont feel strongly on it though, so if you do let me know and i can definitely change it.

Comment (nathanows) [inline spec/integration/admin_items_spec.rb:9]: nm, updated

File: spec/integration/admin_items_spec.rb (hunk 4 above)
Comment (nathanows) [inline spec/integration/admin_items_spec.rb:19]: updated

File: app/views/admin/items/_form.html.erb (hunk 1 above)
Comment (EmilyMB) [inline app/views/admin/items/_form.html.erb:4]: OK. We can go ahead and merge this request (since it's already big). I'll open a ticket to fix it later.

**Comment by mikedao** [comment]
Looks good! I fixed one whitespace issue, no biggie.

**Comment by mikedao** [comment]
I also created an issue to pull unit price out to somewhere else

**Pull request merged**